src/backend/mailbox_manager/api/client/serializers.py

```diff
@@ -36,23 +36,19 @@ def create(self, validated_data):
         Override create function to fire a request on mailbox creation.
         """
         mailbox = super().create(validated_data)
-        if validated_data["domain"].status == enums.MailDomainStatusChoices.ENABLED:
+        if mailbox.domain.status == enums.MailDomainStatusChoices.ENABLED:
             client = DimailAPIClient()
             # send new mailbox request to dimail
-            response = client.create_mailbox(
-                validated_data, self.context["request"].user.sub
-            )
+            response = client.create_mailbox(mailbox, self.context["request"].user.sub)
 
-            # fix format to have actual json, and remove uuid
-            mailbox_data = json.loads(
-                response.content.decode("utf-8").replace("'", '"')
-            )
+            # fix format to have actual json
+            dimail_data = json.loads(response.content.decode("utf-8").replace("'", '"'))
```

Comment:
Why not `response.json()` ? (Question applies not just to this line, but everywhere in the codebase where we have the above form…)

Reply:
because dimail's reponse is not a valid json. That's what I meant by "fix format".

![image](https://github.com/user-attachments/assets/27e96aa9-4bbf-4339-a071-9355b1395e49)

Reply:
Dimail's response seems to be a proper json now so I added it to fixtures and referenced it everywhere